### PR TITLE
HTML-672: Support UUIDs for encounterId parameter when viewing an encounter

### PR DESCRIPTION
**Pull request by ivange94**

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormEntryController.java [hunk 1]
```diff
@@ -107,9 +107,14 @@ else if ("edit".equalsIgnoreCase(modeParam)) {
     	HtmlForm htmlForm = null;
 
     	if (StringUtils.hasText(request.getParameter("encounterId"))) {
-    		
-    		Integer encounterId = Integer.valueOf(request.getParameter("encounterId"));
-    		encounter = Context.getEncounterService().getEncounter(encounterId);
+
+    		String encounterId = request.getParameter("encounterId");
+    		try {
+    			encounter = Context.getEncounterService().getEncounter(Integer.valueOf(encounterId));
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormEntryController.java]: Can you take a look at the formatting?

Comment (ivange94) [reply omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormEntryController.java]: I noticed this. But I executed mvn clean install before committing. Plus formatting looks great inside my IDE.

Comment (ivange94) [reply omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormEntryController.java]: I also checked with vim and formatting looks great.

**Comment by mogoodrich** [comment]
Thanks @ivange94... this looks good, have you tested that everything still works correctly when you pass in an encounterId?

(Not sure what is up with the formatting, but I can take look after the merge and fix if necessary).

Also: not necessary, but in the future it would be good to create a utility method getEncounter(String) in HtmlFormEntryUtil similar to the getters we have for others pieces of data/metadata. (see getConcept, getLocation, etc in HtmlFormEntryUtil).

**Comment by ivange94** [comment]
@mogoodrich yes I tested and works fine. I made a request to /openmrs/module/htmlformentry/htmlFormEntry.form?encounterId=EXISTING_UUID and the encounter was displayed properly.

**Pull request merged**